flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/catalog/FunctionCatalog.java

```diff
@@ -203,39 +201,40 @@ public void registerTempCatalogScalarFunction(ObjectIdentifier oi, ScalarFunctio
 		Set<String> result = getUserDefinedFunctionNames();
 
 		// Get built-in functions
-		result.addAll(
-			BuiltInFunctionDefinitions.getDefinitions()
-				.stream()
-				.map(f -> normalizeName(f.getName()))
-				.collect(Collectors.toSet())
-		);
+		result.addAll(moduleManager.listFunctions());
 
 		return result.toArray(new String[0]);
 	}
 
 	private Set<String> getUserDefinedFunctionNames() {
 		Set<String> result = new HashSet<>();
 
-		// Get functions in catalog
+		// add catalog functions
 		Catalog catalog = catalogManager.getCatalog(catalogManager.getCurrentCatalog()).get();
 		try {
 			result.addAll(catalog.listFunctions(catalogManager.getCurrentDatabase()));
 		} catch (DatabaseNotExistException e) {
 			// Ignore since there will always be a current database of the current catalog
 		}
 
-		// Get functions registered in memory
+		// add temp system functions
 		result.addAll(
 			tempSystemFunctions.values().stream()
 				.map(FunctionDefinition::toString)
 				.collect(Collectors.toSet()));
 
+		// add temp catalog functions
+		result.addAll(
+			tempCatalogFunctions.values().stream()
+				.map(FunctionDefinition::toString)
+				.collect(Collectors.toSet()));
```

Comment:
We should probably move this to be the first as it has higher resolution priority than regular catalog functions.

Reply:
The order shouldn't matter since it only shows a deduped set of names and don't interfere with func resolution. 
Sorry, this actually has nothing to do with code module and I will revert it. Should be done in a separate PR